aten/src/ATen/native/TensorIterator.h

```diff
@@ -175,6 +175,7 @@ struct CAFFE2_API TensorIterator {
   IntArrayRef strides(int arg) const { return operands_[arg].stride_bytes; }
   void* data_ptr(int arg) const;
   ScalarType dtype(int arg=0) const { return operands_[arg].dtype; }
+  ScalarType input_dtype(int arg=0) const { return operands_[num_outputs_ + arg].dtype; }
```

Comment:
What is the use case for writing code to dispatch on the input_dtype? For the use case we were discussing (comparison ops), if we have `torch.eq(tensor, other)`, we want to dispatch on the "common dtype" of (tensor, other), but we will never choose to always dispatch on `tensor`'s dtype or `other`'s dtype.

Reply:
After #22273 commom_dtype will be promoted to other inputs and outputs